specs/ProgrammaticSaveAs.md

```diff
@@ -0,0 +1,422 @@
+COMPLETE. Non-HTML documents, you must use DEFAULT, which will save the content as 
+it is. This API has default values for all parameters, to perform the common 
+save as operation.
+
+# Examples
+## Win32 C++ 
+### Add the Event Handler
+This example hides the default save as dialog and shows a customized dialog.
+```c++
+bool ScenarioSaveAs::AddEventHandler()
+{
+    if (!m_webView2_20)
+        return false;
+
+    // Register a handler for the `SaveAsRequested` event.
+    m_webView2_20->add_SaveAsRequested(
+        Callback<ICoreWebView2SaveAsRequestedEventHandler>(
+            [this](
+            ICoreWebView2* sender,
+                ICoreWebView2SaveAsRequestedEventArgs* args) -> HRESULT
+            {
+                // Hide the system default save as dialog.
+                CHECK_FAILURE(args->put_SuppressDefaultDialog(TRUE));
+                
+                auto showCustomizedDialog = [this, args]
+                {
+                    // Preview the content mime type, optional
+                    wil::unique_cotaskmem_string mimeType;
+                    CHECK_FAILURE(args->get_ContentMimeType(&mimeType));
+
+                    // As an end developer, you can design your own dialog UI, or no UI at all.
+                    // You can ask the user to provide information like file name, file extension, and so on. 
+                    // Finally, and set them on the event args
+                    //
+                    // This is a customized dialog example, the constructor returns after the 
+                    // dialog interaction is completed by the end user.
+                    SaveAsDialog dialog(m_appWindow->GetMainWindow(), saveKinds);
+                    if (dialog.confirmed)
+                    {
+                        // Setting the ResultFilePath, Kind, AllowReplace for the event
+                        // args from this customized dialog inputs is optional.
+                        // The event args has default values based on the document to save.
+                        CHECK_FAILURE(
+                            args->put_ResultFilePath((LPCWSTR)dialog.path.c_str()));
+                        CHECK_FAILURE(args->put_Kind(dialog.selectedKind));
+                        CHECK_FAILURE(args->put_AllowReplace(dialog.allowReplace));
+                    }
+                    else
+                    {
+                        // Save As cancelled from this customized dialog
+                        CHECK_FAILURE(args->put_Cancel(TRUE));
+                    }
+                };
+
+                wil::com_ptr<ICoreWebView2Deferral> deferral;
+                CHECK_FAILURE(args->GetDeferral(&deferral));
+
+                m_appWindow->RunAsync(
+                    [deferral, showCustomizedDialog]()
+                    {
+                        showCustomizedDialog();
+                        CHECK_FAILURE(deferral->Complete());
```

Comment:
In C++ do we need this in a try/finally? Or do we assume the app will go down?

Reply:
No exceptions from Win32 WebView2 APIs.